### PR TITLE
Fix httpconv changes in docs from semconv 1.17 to 1.20

### PR DESCRIPTION
**Pull request by dmathieu**
#4320 upgraded semconv in otelhttp from 1.17.0 to 1.20.0, which has breaking changes:

* `EnduserIDKey` has been removed.
* `HTTPFlavorKey` has been removed.
* `HTTPUserAgentKey` has been renamed to `UserAgentOriginalKey`.

This PR updates the doc comments accordingly.

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

**Comment by dmathieu** [comment]
Ping @Aneurysm9  @hanyuancheung for second review before merge.

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
@MadVikingGod PTAL

**Review by MadVikingGod — APPROVED** [review]

**Pull request merged**
